MIP/mip-58/README.md

```diff
@@ -0,0 +1,147 @@
+
+## Specification
+
+The simplified bridge design focuses on minimizing complexity and maximizing security:
+
+### Key Features
+
+1. **Two-Transaction Model**:
+   - **Initiation**: User sends a transaction to initiate the bridge.
+   - **Completion**: A relayer or multi-signature group completes the transfer on the counterparty contract.
+
+2. **Consolidation of Logic**:
+   - Merge lock and completion functionality on the counterparty contract. Once lock is called, funds are already in the control of the user. Once the timelock is over and complete on initiator has not been callde, both the initiator and counterparty funds are available to the user.
+   - Remove refund functionality entirely to eliminate associated exploits.
+
+3. **Parameter Validation**:
+   - Ensure parameter validation on the counterparty to prevent invalid transactions.
+
+4. **Relayer Redundancy**:
+   - Use two types of relayers:
+     - **Automated Relayer**: Operated with minimal human involvement; its private key is highly secured.
+     - **Multi-Signature Relayer**: Managed by the team to guarantee transaction completion in case of failures.
+
+5. **Cost Efficiency**:
+   - Minimize gas costs by reducing the number of interactions and simplifying fee calculations.
+
+6. **Enhanced Security**:
+   - Avoid refund logic to close exploit windows.
+   - Protect against key compromise through key isolation or known by no parties and multi-signature relayer setups.
+   - There is no scenario where a bridge could lead to double-spending. It's either completed by relayers or not.
+
+7. **Batch Processing for Downtime**:
+   - Multi-signature relayers can process multiple pending transactions in a single batch during downtime.
+
+8. **BridgeTransferId**:
+   - Continue using unique identifier to prevent double-spending and track transactions securely.
+
+9. **Best Practices**:
+   - Adopt currently used bridge designs from established designs like Arbitrum, LayerZero and Blast bridges which use a relayer to finalize the bridge.
+   - User is not required to have funds on counterparty contract to finalize the bridge.
+
+
+## Exploits and Potential Losses
+
+1. **Rate-Limit Exploits**:
+   - If the relayer is down, an attacker could exploit refund functionality to repeatedly bridge funds within the rate limit.
+2. **Key Compromise**:
+   - Relayer or refund keys could be compromised, leading to unauthorized transactions.
+3. **Fee Misestimation**:
+   - Incorrect fee calculations (e.g., underestimating gas) can cause significant financial losses for the protocol.
+4. **User Abandonment**:
+   - High costs and complexity deter users, leading to lower adoption and reputational damage.
+
+## Reference Implementation
+
+1. **Two-Transaction Flow**:
+   - User initiates the transfer.
+   - Relayer or multisig completes the transfer with parameter validation.
```

Comment:
Could you add something about how users are supposed to be able to check on-chain that their transfer is complete? Does the relayer keep some database where users can look up transactions associated with a specific bridge transfer ID?